packages/trade-widget/src/components/TokenSelector/TokenSelector.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+    Button,
+    Column,
+    Columns,
+    Content,
+    Control,
+    Dropdown,
+    DropdownContent,
+    DropdownDivider,
+    DropdownItem,
+    DropdownMenu,
+    DropdownTrigger,
+    Field,
+    Icon,
+    Image,
+    Label,
+    Select,
+} from 'bloomer';
+import * as _ from 'lodash';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import { AssetToken } from '../../types';
+
+import TokenSelectorItem from '../TokenSelectorItem';
+
+import { definedTokensToIcon } from '../TokenSelectorItem/TokenSelectorItem';
+
+interface TokenSelectorPropTypes {
+    onChange?: (token: AssetToken) => void;
+    selectedToken: AssetToken;
+}
+
+interface TokenMetadata {
+    id: string;
+    symbol: AssetToken;
+    description: string;
+}
+class TokenSelector extends React.Component<TokenSelectorPropTypes> {
+    // tslint:disable-next-line:underscore-private-and-protected
+    private static defaultProps = {
+        // tslint:disable-next-line:no-empty
+        onChange: () => {},
+    };
+    private _tokens: TokenMetadata[];
+    constructor(props: TokenSelectorPropTypes) {
+        super(props);
+        this.handleItemSelected = this.handleItemSelected.bind(this);
+        this._tokens = [
+            { symbol: AssetToken.ZRX, id: AssetToken.ZRX, description: '0x Token' },
+            { symbol: AssetToken.BAT, id: AssetToken.BAT, description: 'Basic Attention Token' },
```

Comment:
Symbol and id seem duplicitous. What's the difference? Also, should be by relying on symbols as the id for tokens? They are not guaranteed to be unique. Really we should be using their contract addresses.